allennlp/data/dataset_readers/text_classification_json.py

```diff
@@ -0,0 +1,120 @@
+from typing import Dict, List
+import logging
+import json
+from overrides import overrides
+from allennlp.common.file_utils import cached_path
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import LabelField, TextField, Field, ListField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Tokenizer, WordTokenizer
+from allennlp.data.tokenizers.sentence_splitter import SpacySentenceSplitter
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("text_classification_json")
+class TextClassificationJsonReader(DatasetReader):
+    """
+    Reads tokens and their labels from a labeled text classification dataset.
+    Expects a "tokens" field and a "category" field in JSON format.
+
+    The output of ``read`` is a list of ``Instance`` s with the fields:
+        tokens: ``TextField`` and
+        label: ``LabelField``
+
+    Parameters
+    ----------
+    token_indexers : ``Dict[str, TokenIndexer]``, optional
+        optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We use this to define the input representation for the text.
+        See :class:`TokenIndexer`.
+    tokenizer : ``Tokenizer``, optional (default = ``{"tokens": WordTokenizer()}``)
+        Tokenizer to use to split the input text into words or other kinds of tokens.
+    segment_sentences: ``bool``, optional (default = ``False``)
+        If True, we will first segment the text into sentences using SpaCy and then tokenize words.
+        Necessary for some models that require pre-segmentation of sentences,
+        like the Hierarchical Attention Network.
+    sequence_length: ``int``, optional (default = ``None``)
```

Comment:
maybe use `max_sequence_length` instead? I think this makes it consistent with other pieces of code in the library.